BlueShift-iOS-SDK/BlueShiftDeviceData.m

```diff
@@ -151,21 +151,23 @@ - (NSDictionary *)toDictionary {
 -(void)saveDeviceDataForNotificationExtensionUse {
     //Save device data for notifcation extension to use it and send to server with delivered tracking api call
     if (@available(iOS 10.0, *)) {
+        NSString *deviceId = self.deviceUUID;
+        NSString *appName = [[BlueShiftAppData currentAppData] bundleIdentifier];
         NSString *appGroupID = [BlueShift sharedInstance].config.appGroupID;
-        if(appGroupID && ![appGroupID isEqualToString:@""]) {
+        
+        if(appGroupID && ![appGroupID isEqualToString:@""] && deviceId && appName) {
```

Comment:
what will happen once the deviceID & appName coming as empty..?

Reply:
Empty value will be saved.

Reply:
okay. got it.

Reply:
if someone does not provide us with a appGroupID, this block of code will not execute, right? will that cause any issue?

Reply:
If any of appGroupId/ deviceid/ appName is nil or appGroupId is empty string, then it will not execute this block.